packages/gator-permissions-snap/src/ui/components/Field.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  Box,
+  Text,
+  Button,
+  Field as SnapField,
+  Icon,
+} from '@metamask/snaps-sdk/jsx';
+import type { SnapElement } from '@metamask/snaps-sdk/jsx';
+
+import { TokenIcon } from './TokenIcon';
+import { TooltipIcon } from './TooltipIcon';
+
+export type BaseFieldProps = {
+  label: string;
+  tooltip?: string | undefined;
+  errorMessage?: string | undefined;
+  disabled?: boolean | undefined;
+  iconData?:
+    | {
+        iconDataBase64: string;
+        iconAltText: string;
+      }
+    | undefined;
+  removeButtonName?: string | undefined;
+  children: SnapElement;
+  variant?: 'form' | 'display' | undefined;
+};
+
+export const Field = ({
+  label,
+  tooltip,
+  errorMessage,
+  disabled,
+  iconData,
+  removeButtonName,
+  children,
+  variant = 'form',
+}: BaseFieldProps) => {
+  const iconElement = iconData ? (
+    <TokenIcon
+      imageDataBase64={iconData.iconDataBase64}
+      altText={iconData.iconAltText}
+    />
+  ) : null;
+
+  const tooltipElement = tooltip ? <TooltipIcon tooltip={tooltip} /> : null;
+
+  const removeButtonElement =
+    removeButtonName && !disabled ? (
+      <Button name={removeButtonName} type="button">
+        <Icon name="close" color="primary" size="md" />
+      </Button>
+    ) : null;
+
+  const labelSection = (
+    <Box direction="horizontal">
+      <Text>{label}</Text>
+      {tooltipElement}
+    </Box>
+  );
+
+  if (variant === 'display') {
+    return (
+      <Box direction="horizontal" alignment="space-between">
+        {labelSection}
+        <Box direction="horizontal">
+          {iconElement}
+          {children}
+        </Box>
+      </Box>
+    );
+  }
+
+  return (
+    <Box direction="vertical">
+      <Box direction="horizontal" alignment="space-between">
+        {labelSection}
+      </Box>
+      {children.type === 'Dropdown' ? (
+        <SnapField error={errorMessage}>{children as JSX.Element}</SnapField>
+      ) : (
+        <SnapField error={errorMessage}>
+          <Box>{iconElement}</Box>
+          {children as JSX.Element}
+          <Box>{removeButtonElement}</Box>
+        </SnapField>
```

Comment:
Can we simplify this so that all field types render the same? One of the things that I don't like about the status quo _before_ this PR is that different fields behave differently (for instance you can't have a dropdown field that's optional, even though you can (iirc) specify it on the field definition.

It would be super nice to have all fields behave exactly the same.